### PR TITLE
PLT-976 Adds ability to sign in with username

### PR DESCRIPTION
**Pull request by rgarmsen2295**
Adds a login field to the login screen that accepts username (similar to the pre-existing ldap and email versions) if the requisite config setting is marked as true.  Also separates the config setting for enabling email sign-up and sign-in so that users can still sign-up with email while disabling email login (and only enabling username login).

### CODE REVIEW

File: api/user.go [hunk 1]
```diff
@@ -443,6 +443,37 @@ func LoginByEmail(c *Context, w http.ResponseWriter, r *http.Request, email, nam
...
+
+	if result := <-Srv.Store.User().GetByUsername(team.Id, username); result.Err != nil {
+		c.Err = result.Err
+		c.Err.StatusCode = http.StatusForbidden
+		return nil
+	} else {
+		user := result.Data.(*model.User)
+
+		if len(user.AuthData) != 0 {
+			c.Err = model.NewAppError("LoginByUsername", "Please sign in using "+user.AuthService, "")
```
Comment (coreyhulen) [inline api/user.go]: All errors should now be using `NewLocAppError` and all logging should use `utils.T()`

**Comment by rgarmsen2295** [comment]
Ok @coreyhulen changes complete, please take another look when possible

**Comment by enahum** [comment]
if #2002 is merge with master the settings in system console should use loc as the server side

**Comment by rgarmsen2295** [comment]
Sounds good, I figured I would probably have to go back and edit to account for the loc changes sooner rather than later :) I'll edit as necessary once the loc stuff is merged in

**Comment by hmhealey** [comment]
@rgarmsen2295 I just merged the system console localization stuff if you want to go through and update your stuff to be localized now

**Comment by rgarmsen2295** [comment]
Thanks @hmhealey, I'll change it over next time I'm in

**Comment by rgarmsen2295** [comment]
Okay changes complete @hmhealey @coreyhulen please take a look when possible, pretty sure I hit all the necessary loc changes but not sure if there's a different process (e.g. I didn't include Spanish translations for my new strings)

**Comment by rgarmsen2295** [comment]
Rebase complete @jwilander.  Also I forgot to mention in the description that this has some config file changes as well.

**Comment by coreyhulen** [comment]
+1

**Comment by jwilander** [comment]
+1

**Pull request merged**